runtime/vm/createramclass.cpp

```diff
@@ -2803,14 +2803,10 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 					module = javaVM->unamedModuleForSystemLoader;
 				}
 			} else {
-				if (J9ROMCLASS_IS_PRIMITIVE_TYPE(romClass)
-					|| (LOAD_LOCATION_PATCH_PATH == locationType)
-					|| (LOAD_LOCATION_MODULE == locationType)
-				) {
-					module = javaVM->javaBaseModule;
-				} else {
-					module = javaVM->unamedModuleForSystemLoader;
-				}
+				/* Do not assign module based on locationType, as we never set locationType for anonymous classes which led us to assign them to
+				 * unamedModuleForSystemLoader incorrectly. Change the code to directly assign classes to Java.base here.
+				 */
```

Comment:
This isn't quite the comment I was expecting as the situation is more broad than anonymous classes, which are correctly assigned the `hostClass->module` after java.base is created.

Maybe something more along the lines of:
```
Ignore locationType and assign all classes created before the java.base module is created to java.base.
This matches the RI implementation.  Validated on JDK9 through JDK11.
```